### DISCUSSION TITLE
Recreate existing cluster

### DISCUSSION THREAD

**Discussion by viceice** [longhorn/longhorn#2189] [General]
I'm currently using a k3s v1.20 cluster in integrated etcd mode and like to recreate it in mysql / mariadb mode.
I'm using nfs for longhorn volume backup.
I can recreate all my infrastructure by simply applying my terraform configurations.

Whats the best way to recreate the longhorn cluster and volumes in a new k3s cluster with existing `/var/lib/longhorn` directory?
Where do i find the longhorn cluster configuration to reapply to the new cluster?

A downtime of the k3s cluster is no problem.

**Comment by joshimoo** [comment]
cc @shuo-wu @meldafrawi

**Comment by shuo-wu** [comment]
Right now there is no convenient way to snapshot and backup the whole Longhorn system configuration as well as building a volume based on an existing data directory. I know Rancher can do snapshots and restorations for the whole cluster including Longhorn but it's not controlled by Longhorn itself. And we have a way to start a volume manually based on a replica directory but it's tricky: https://github.com/longhorn/longhorn-engine#running-a-controller-with-multiple-replicas

@yasker Do we have plans for this?

**Reply by viceice** [reply]
For me it would be enough to know what rancher will backup and restore, so i can to this maually too

**Comment by yasker** [comment]
We have an issue tracking the effort to backup the longhorn system at https://github.com/longhorn/longhorn/issues/1455 . Currently scheduling for v1.2.0 release.
